### PR TITLE
fix: [DevOps] Fix Broken Maven Plugin Tests and Ban JUnit4

### PR DESCRIPTION
**Pull request by MatKuhr**
## Overview

This pull request fixes the tests for our 3 maven plugins. Currently, the tests are **not executed**. This is because they are defined using JUnit 4, but our JUnit execution engine (defined here) only executes JUnit 5 tests. Unfortunately, this goes completely undetected.

## Changes

* Migrate maven plugin tests to JUnit 5
  * Apply a bit of workaround code to make the JUnit4-based `MojoRule` from `maven-plugin-test-harness` work inside the JUnit 5 tests
  * Fix any broken test code
* Ban JUnit 4 via dependency rule
* ~Ban import statements for JUnit 4 test annotation (`org.junit.Test`) via PMD rule~ 
  * Moved to separate PR
  * This is done to enforce that in the maven plugin modules, where we still need to have JUnit 4 on the classpath, it can't be used to annotate tests

## Notes

Alternatively to:

> Migrate maven plugin tests to JUnit 5

one could also change the JUnit execution engine just for the maven plugin modules. But that also kinda feels bad and again, the risk is a JUnit 5 test is added but not executed.

So I opted for the workaround code for now..

### CODE REVIEW

File: cloudplatform/security/src/test/java/com/sap/cloud/sdk/cloudplatform/security/principal/OAuth2AuthTokenPrincipalExtractorTest.java [hunk 1]
```diff
@@ -49,13 +48,6 @@ class OAuth2AuthTokenPrincipalExtractorTest
                 PREFIX + ".Persons.View",
                 PREFIX + ".Workplaces.View");
 
-    private static final List<String> PREFIX_STRIPPED_SCOPES =
```
Comment (MatKuhr) [inline cloudplatform/security/src/test/java/com/sap/cloud/sdk/cloudplatform/security/principal/OAuth2AuthTokenPrincipalExtractorTest.java:52]: driveby dead code removal

File: datamodel/openapi/openapi-generator-maven-plugin/src/test/java/com/sap/cloud/sdk/datamodel/openapi/generator/DataModelGeneratorMojoIntegrationTest.java [hunk 2]
```diff
@@ -54,25 +52,17 @@ public void generateAndCompareSodastoreLibrary()
     }
 
     // Run this test method manually to overwrite the folder containing the expected content with the latest generator state
-    //@Test
-    public void regenerateExpectedSodastoreLibrary()
-        throws Exception
+    @Test
```
Comment (MatKuhr) [inline datamodel/openapi/openapi-generator-maven-plugin/src/test/java/com/sap/cloud/sdk/datamodel/openapi/generator/DataModelGeneratorMojoIntegrationTest.java:55]:
```suggestion
    // @Test
```

File: codestyle/sdk_specific_pmd_rules.xml [hunk 3]
```diff
@@ -122,6 +122,29 @@
             ]]>
         </example>
     </rule>
+    <rule name="UseOfJunit4" language="java"
```
Comment (MatKuhr) [inline codestyle/sdk_specific_pmd_rules.xml:125]: the rule doesn't seem to work quite yet, trying to fix it

File: cloudplatform/connectivity-destination-service/pom.xml [hunk 4]
```diff
@@ -176,6 +176,12 @@
 			<groupId>com.sap.cloud.security</groupId>
 			<artifactId>java-security-test</artifactId>
 			<scope>test</scope>
+			<exclusions>
+				<exclusion>
+					<artifactId>junit</artifactId>
+					<groupId>junit</groupId>
+				</exclusion>
+			</exclusions>
```
Comment (newtork) [inline cloudplatform/connectivity-destination-service/pom.xml:184]: (Comment)

I was wondering why not add this to dependency management, unfortunately it's coming in transitively via BOM. :)

File: pom.xml [hunk 5]
```diff
@@ -165,7 +166,7 @@
 			<dependency>
 				<groupId>org.apache.maven.plugin-tools</groupId>
 				<artifactId>maven-plugin-annotations</artifactId>
-				<version>${maven-plugin.version}</version>
+				<version>3.11.0</version>
```
Comment (newtork) [inline pom.xml:169]: (Minor)

Please add a property for this version next to the others :)

**Review by newtork — APPROVED** [review]

File: codestyle/sdk_specific_pmd_rules.xml (hunk 3 above)
Comment (MatKuhr) [reply codestyle/sdk_specific_pmd_rules.xml]: split out to separate PR

**Review by newtork — APPROVED** [review]
